src/thx/promise/Promise.hx

```diff
@@ -25,24 +26,17 @@ abstract Promise<T>(Future<Result<T, Error>>) to Future<Result<T, Error>> {
 
   public static var nil(default, null) : Promise<Nil> = Promise.value(Nil.nil);
 
-  public static function sequence(arr : Array<Promise<Dynamic>>) : Promise<Nil>
-    return Promise.create(function(resolve : Dynamic -> Void, reject) {
-      arr = arr.copy();
-      function poll() {
-        if(arr.isEmpty()) {
-          resolve(nil);
-        } else {
-          arr.shift()
-            .success(function(_) poll())
-            .failure(reject);
+  public static inline function wrapEffect<A, B>(f: A -> B): A -> Promise<B>
+    return function(a: A) return Promise.create(function(resolve, reject) {
+      try {
+        resolve(f(a));
+      } catch (e: Dynamic) {
+        try {
+          reject(Error.fromDynamic(e));
+        } catch (e: Dynamic) {
+          reject(new thx.Error("Unrecoverable error; could not convert dynamic value to error type."));
```

Comment:
this is using the same faulty `reject` ... it can easily lead to a loop ... also not sure why `reject(Error.fromDynamic(e));` would not be ok here